File: main-svc/src/main/java/com/chilly/main_svc/model/Visit.java
package com.chilly.main_svc.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.proxy.HibernateProxy;

import java.util.Date;
import java.util.Objects;

@Entity(name = "visit")
@Table(name = "visits")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Visit {

    @Id
    @SequenceGenerator(name = "visit_sequence", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "visit_sequence")
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "place_id")
    private Place place;

    @Column(name = "date")
    private Date date;

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (o == null) return false;
        Class<?> oEffectiveClass = o instanceof HibernateProxy ? ((HibernateProxy) o).getHibernateLazyInitializer().getPersistentClass() : o.getClass();
        Class<?> thisEffectiveClass = this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass() : this.getClass();
        if (thisEffectiveClass != oEffectiveClass) return false;
        Visit visit = (Visit) o;
        return getId() != null && Objects.equals(getId(), visit.getId());
    }

    @Override
    public final int hashCode() {
        return this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass().hashCode() : getClass().hashCode();
    }
}

File: places-svc/src/main/kotlin/com/chilly/places_svc/model/Visit.kt
package com.chilly.places_svc.model

import jakarta.persistence.*
import org.hibernate.Hibernate
import java.util.*

@Entity(name = "visit")
@Table(name = "visits")
class Visit (
    @Column(name = "date", nullable = false)
    var date: Date
) {
    @Column(name = "user_id", nullable = false)
    var userId: Long = -1

    @ManyToOne
    @JoinColumn(name = "place_id", nullable = false)
    lateinit var place: Place

    @Id
    @SequenceGenerator(name = "visit_sequence", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "visit_sequence")
    @Column(name = "id")
    var id: Long? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as Visit

        return id == other.id
    }

    override fun hashCode(): Int = javaClass.hashCode()
}